mint-client/src/clients/gateway.rs

```diff
@@ -226,75 +229,27 @@ impl GatewayClient {
+        outpoint: OutPoint,
+    ) -> Result<()> {
+        self.context
+            .api
+            .await_output_outcome::<OutgoingContractOutcome>(outpoint, Duration::from_secs(10))
+            .await?;
+        // We remove the entry that indicates we are still waiting for transaction
+        // confirmation. This does not mean we are finished yet. As a last step we need
+        // to fetch the blind signatures for the newly issued tokens, but as long as the
+        // federation is honest as a whole they will produce the signatures, so we don't
+        // have to worry
+        self.context
             .db
-            .get_value(&OutgoingPaymentClaimKey(contract_id))
-            .expect("DB error")
-            .expect("Contract not found");
-        let txid = transaction.tx_hash();
-
-        let await_confirmed = || async {
-            loop {
-                match self.context.api.fetch_tx_outcome(txid).await {
-                    Ok(TransactionStatus::Accepted { .. }) => return,
-                    Ok(TransactionStatus::Error(e)) => {
-                        panic!("Transaction was rejected by federation: {}", e);
-                    }
-                    Err(e) => {
-                        if e.is_retryable_fetch_coins() {
-                            tokio::time::sleep(tokio::time::Duration::from_secs(1)).await;
-                        } else {
-                            // FIXME: maybe survive a few of these and only bail if the error remains, federation could be unreachable
-                            panic!(
-                                "Federation returned error when fetching tx outcome: {:?}",
-                                e
-                            );
-                        }
-                    }
-                }
-            }
-        };
-
-        let timeout = tokio::time::Duration::from_secs(10);
-        for _ in 0..6 {
-            tokio::select! {
-                () = await_confirmed() => {
-                    // We remove the entry that indicates we are still waiting for transaction
-                    // confirmation. This does not mean we are finished yet. As a last step we need
-                    // to fetch the blind signatures for the newly issued tokens, but as long as the
-                    // federation is honest as a whole they will produce the signatures, so we don't
-                    // have to worry about that.
-                    self.context
-                        .db
-                        .remove_entry(&OutgoingPaymentClaimKey(contract_id))
-                        .expect("DB error");
-                    return;
-                },
-                _ = tokio::time::sleep(timeout) => {
-                    self.context
-                        .api
-                        .submit_transaction(transaction.clone())
-                        .await
-                        .expect("Error submitting transaction");
-                }
```

Comment:
I dropped this. It seemed confusing and perhaps premature optimization. Should I add back?